lib/CArtHandler.cpp

```diff
@@ -419,8 +419,8 @@ ArtifactPosition::ArtifactPosition(std::string slotName):
 	auto it = artifactPositionMap.find (slotName);
 	if (it != artifactPositionMap.end())
 		num = it->second;
-
-	logMod->warn("Warning! Artifact slot %s not recognized!", slotName);
+	else

```

Comment:
@rilian-la-te 
After method was converted to constructor, code
` return it->second`
became
`num = it->second`

As result, method no longer returns & next line executes unconditionally

Reply:
also missed it :(

Reply:
And I missed it too( Sorry(